api/jobs/jobs.py

```diff
@@ -83,10 +125,14 @@ def load(cls, e):
         # Don't modify the map
         d = copy.deepcopy(e)
 
-        if d.get('inputs', None):
-            inputs = d['inputs']
-            for x in inputs.keys():
-                inputs[x] = create_filereference_from_dictionary(inputs[x])
+        if d.get('inputs'):
+            input_dict = {}
+
+            for i in d['inputs']:
```

Comment:
I realize I didn't explain what the intent was. With this conversion we can maintain a job structure in Mongo that makes lookups on inputs efficient but can also have a job structure via a Python obj that maintains uniqueness of input keys and allows us to restructure the mongo schema without a bit refactor.